pyasic/miners/backends/luxminer.py

```diff
@@ -146,6 +147,32 @@ async def reboot(self) -> bool:
     async def get_config(self) -> MinerConfig:
         return self.config
 
+    async def upgrade_firmware(self) -> dict:
+        """
+        Upgrade the firmware on a LuxOS miner by calling the 'updaterun' API command.
+
+        Returns: Result of the upgrade process.
+
+        """
+        command = "updaterun"
+
+        try:
+            response = await self.rpc.send_command(command=command)
+
+            if not response:
+                raise ValueError("No response received during the firmware upgrade process.")
+
+            if response.get("status") == "success":
+                logging.info(f"{self.ip}: Firmware upgrade initiated successfully.")
+                return "Firmware upgrade completed successfully."
+            else:
+                logging.error(f"{self.ip}: Firmware upgrade failed. Response: {response}")
+                raise ValueError(f"Firmware upgrade failed. Response: {response}")
+
+        except Exception as e:
```

Comment:
Catch `APIError` here instead, it will automatically get raised if there is an issue, or if the command fails.

Reply:
okay